File: library/src/androidTest/java/com/grosner/dbflow/test/sql/TestModel3.java
package com.grosner.dbflow.test.sql;

import com.grosner.dbflow.annotation.Column;
import com.grosner.dbflow.annotation.ContainerAdapter;
import com.grosner.dbflow.annotation.Table;
import com.grosner.dbflow.test.TestDatabase;
import com.grosner.dbflow.test.structure.TestModel1;

/**
* Author: andrewgrosner
* Contributors: { }
* Description:
*/
@Table(databaseName = TestDatabase.NAME)
@ContainerAdapter
public class TestModel3 extends TestModel1 {
    @Column
    String type;
}

File: dbflow-tests/src/test/java/com/raizlabs/android/dbflow/structure/TestModel3.kt
package com.raizlabs.android.dbflow.structure

import com.raizlabs.android.dbflow.TestDatabase
import com.raizlabs.android.dbflow.annotation.*
import java.sql.Date

@Table(database = TestDatabase::class)
class TestModel3 : BaseModel() {

    @Column
    @PrimaryKey
    var name: String? = null

    @Column(length = 5, collate = Collate.BINARY)
    var order: Int = 0

    @Column(collate = Collate.NOCASE)
    var date: Date? = null

    @Column
    @ForeignKey(references = arrayOf(ForeignKeyReference(columnName = "testAI_name",
        foreignKeyColumnName = "name")))
    var testAutoIncrement: TestModel1? = null
}
